libraries/shared/src/SpatiallyNestable.cpp

```diff
@@ -1171,9 +1171,39 @@ bool SpatiallyNestable::updateQueryAACube() {
 
     _queryAACubeSet = true;
 
-    auto parent = getParentPointer(success);
-    if (success && parent) {
-        parent->updateQueryAACube();
+    if (updateParent) {
+        auto parent = getParentPointer(success);
+        if (success && parent) {
+            parent->updateQueryAACube();
+        }
+    }
+
+    return true;
+}
+
+bool SpatiallyNestable::updateQueryAACubeWithDescendantAACube(const AACube& descendantAACube, bool updateParent) {
```

Comment:
this and the below method are identical to the existing methods, but they don't need to loop over all their descendants